### PR TITLE
ONXP-763 [Account settings] Add the button Delete my user account

### PR DESCRIPTION
**Pull request by OlehHinko**

### CODE REVIEW

**Comment by OlehHinko** [comment]
@goonerDp, please check

File: src/api/profile.js [hunk 1]
```diff
@@ -21,3 +21,9 @@ export async function changeProfile(data) {
 	const form = await restClient.put("/profile", data, { headers: { ...authHeaders } });
 	return form;
 }
+
+export async function deleteUserAccount() {
+	const authHeaders = getAuthHeaders();
+	const form = await restClient.delete("/profile", { headers: { ...authHeaders } });
+	return form;
```
Comment (goonerDp) [inline src/api/profile.js]: Why is this variable called "form"?

File: src/pages/profile/DeleteAccount.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const { confirm } = Modal;
+
+function DeleteAccount(props) {
+	const [isLoadingDeleteAccount, toggleBtnLoading] = useState(false);
+	const { logOut } = props;
+	const showMessage = () => {
+		toggleBtnLoading(true);
+		confirm({
+			title: "Are you sure delete your accounts?",
```
Comment (goonerDp) [inline src/pages/profile/DeleteAccount.js]: typo

File: src/pages/profile/DeleteAccount.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+			content:
+				"After deleting an account,  you will lose access to all their assets, and to restore access, you will need to register a new account with the same wallet.",
```
Comment (goonerDp) [inline src/pages/profile/DeleteAccount.js]: After deleting an account,  you will lose access to all your assets, and to restore access, you will need to register a new account with the same wallet.

File: src/api/profile.js (hunk 1 above)
Comment (goonerDp) [reply src/api/profile.js]: What about errors handling?

File: src/pages/profile/DeleteAccount.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+			okText: "Yes",
+			okType: "danger",
+			cancelText: "No",
+			async onOk() {
+				const res = await deleteUserAccount();
+				if (res.data === "OK") {
+					await logOut();
+					showNotification({
```
Comment (goonerDp) [inline src/pages/profile/DeleteAccount.js]: After logout, the app will refresh, and the notification will not be shown.

File: src/pages/profile/DeleteAccount.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	return (
+		<DeleteAccountContainer>
+			<Button
+				type="danger"
+				loading={isLoadingDeleteAccount}
+				className="btn-delete-account"
+				onClick={() => showMessage()}
+			>
+				Delete my user account
+			</Button>
```
Comment (goonerDp) [inline src/pages/profile/DeleteAccount.js]: You can use Popconfirm, directly
https://ant.design/components/popconfirm/#components-popconfirm-demo-dynamic-trigger

**Pull request merged**